lib/Twig/Template.php

```diff
@@ -126,6 +126,34 @@ public function displayParentBlock($name, array $context, array $blocks = array(
     }
 
     /**
+     * Checks if a block exists.
+     *
+     * This method is for internal use only and should never be called
+     * directly.
+     *
+     * @param string $name      The block name to test
+     * @param array  $context   The context
+     * @param array  $blocks    The current set of blocks
+     * @param bool   $useBlocks Whether to use the current set of blocks
+     *
+     * @return bool
+     */
+    public function existsBlock($name, array $context, array $blocks = array(), $useBlocks = true)
+    {
+        $name = (string) $name;
```

Comment:
should we really cast here ? IMO, we should expect a string here all the time. The casting could be done in the compiled code of the ``Twig_Node_Expression_Test_Block``. It would avoid casting again when calling the method on the parent